deeptiling/deep_tiling_models.py

```diff
@@ -0,0 +1,642 @@
+                try: 
+                    pca_obj = self.best_results['PCA_transformer']
+                    embs = pca_obj.fit_transform(embs)
+                except AttributeError:
+                    print('Warning: PCA transformation was not previously computed\
+                          it will be computed directly on the current sentences')
+                          
+                    if pca_components is None:
+                        print("Warning: provide number of principal components\
+                              in order to perform PCA. No such number has been\
+                              provided: PCA won't be performed.")
+                    
+                    else:
+                        pca_obj = PCA(n_components = pca_components)
+                    
+                        embs = pca_obj.fit_transform(embs)
+              
+              embs = pd.DataFrame(embs)
+              
+              if C99:
+                  boundaries = np.array(self.C99(embs, std_coeff = C99_std_coeff, window = window, 
+                  transform_rank = C99_rank_transform))
+  
+            if not C99:
+                scores, depth_scores = self.compute_depth_score(embs, 
+                                                              window = window, 
+                                                              clip=clip,
+                                                              single = multi_encode,
+                                                              smooth = smooth)
+                
+                boundaries = self.compute_boundaries(depth_scores, threshold=(np.mean(depth_scores) +
+                                            np.std(depth_scores)*threshold), number_of_segments = number_of_segments)
+            
+            if timer:
+              end = time.time()
+              times = end-start
+              print("Segmentation completed in {} minutes".format(times))
+        
+        else:
+            raise Exception('Provide a non-empty list of sentences!')
+            
+        segments = []
+        segmented_embs = []
+        last_index = 0
+        for index, boundary in enumerate(boundaries):
+            if boundary:
+                segments.append(sentences[last_index:index + 1])
+                segmented_embs.append(embs.iloc[last_index:index + 1].values)
+                last_index = index + 1
+        
+        segments.append(sentences[last_index:])
+        segmented_embs.append(embs.iloc[last_index:].values)
+        if C99:
+            return {'segments': segments, 
+                    'boundaries': boundaries,
+                    'similarity_matrix': self.rank,
+                    'embeddings': segmented_embs}
+        else:
+            return {'segments': segments, 
+                    'boundaries': boundaries,
+                    'depth_scores': depth_scores,
+                    'embeddings': segmented_embs,
+                    "scores": [0.0] + scores} 
```

Comment:
I believe you've added this line here, maybe the only change to the original. I would prefer that we don't change this module at all. This is  for "forwards compatability" reasons, i.e. that I can update it in future if the upstream changes simply by copy-pasting the whole file; or that I can make the upstream an actual dependency if it ever declares itself porperly.

In order to have no changes to this module, I think we need to either:
(a) call compute_depth_scores instead of predict; or 
(b) extend the DeepTiling class and override the compute_boundaries method

I'm not sure which is best, I quite like the idea of the second approach.

Reply:
Nice work with the child class